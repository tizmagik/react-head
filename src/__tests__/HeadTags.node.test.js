import * as React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import { HeadProvider, Title, Style, Meta, Link } from '../';

describe('head tag during server', () => {
  it('renders nothing and adds tags to headTags context array', () => {
    const arr = [];
    const globalCss = `p {
      color: #121212;
    }`;
    const markup = renderToStaticMarkup(
      <HeadProvider headTags={arr}>
        <div>
          Yes render
          <Title>Title</Title>
          <Style>{globalCss}</Style>
          <Link href="index.css" />
          <Meta charset="utf-8" />
        </div>
      </HeadProvider>
    );
    expect(markup).toMatchSnapshot();
    expect(arr).toMatchSnapshot();
  });

  it('renders only the last title', () => {
    const arr = [];
    renderToStaticMarkup(
      <HeadProvider headTags={arr}>
        <div>
          <Title>Title 1</Title>
        </div>
        <div>
          <Title>Title 2</Title>
        </div>
        <div>
          <Title>Title 3</Title>
        </div>
      </HeadProvider>
    );
    expect(arr).toMatchSnapshot();
  });

  it('renders only the last meta with the same name', () => {
    const arr = [];
    renderToStaticMarkup(
      <HeadProvider headTags={arr}>
        <Meta>Meta 1</Meta>
        <Meta>Meta 2</Meta>
        <Meta name="1">Meta 3</Meta>
        <Meta name="2">Meta 4</Meta>
        <Meta name="3">Meta 5</Meta>
        <Meta name="2">Meta 6</Meta>
      </HeadProvider>
    );
    expect(arr).toMatchSnapshot();
  });
});
