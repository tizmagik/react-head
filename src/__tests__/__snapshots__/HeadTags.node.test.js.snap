// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadTag during server rendering adds tags to headTags context array 1`] = `
Array [
  <test
    content="testing tag"
    data-rh=""
  >
    No render
  </test>,
]
`;

exports[`HeadTag during server rendering renders nothing 1`] = `"<div>Yes render</div>"`;
